cmake/iDynTreeOptions.cmake

```diff
@@ -131,3 +131,18 @@ macro(idyntree_enable_cxx11)
   endif()
 endmacro()
 
+# Macro to remove deprecation warnings for a specific part of iDynTree
+macro(idyntree_disable_deprecation_warnings)
+  if(MSVC)
+    add_definitions(/wd4996)
+  else()
+    check_cxx_compiler_flag("-Wno-deprecated-declarations" CXX_HAS_WNO_DEPRECATED_DECLARATIONS)
+    if(CXX_HAS_WNO_DEPRECATED_DECLARATIONS)
+      add_definitions(-Wno-deprecated-declarations)
+    endif()
+    check_cxx_compiler_flag("-Wno-deprecated" CXX_HAS_WNO_DEPRECATED)
+    if(CXX_HAS_WNO_DEPRECATED_DECLARATIONS)
```

Comment:
This `if` is wrong. It should be `if(CXX_HAS_WNO_DEPRECATED)`

Reply:
Good catch, fixed now.

Reply:
Fixed in https://github.com/robotology/idyntree/pull/358/commits/cef7633fe96242ed3cc4160a96c9c6fcda3b24b5 .